Issue 31 elevator commands

SuzakuSar opened the pull request:
Added 6 commands for the 4 levels of the bot and 2 for while held down for testing

SuzakuSar committed:
Added elevatorconstants and 6 commands
4 positions and 2 voltage

SuzakuSar committed:
Added comments to commands

SuzakuSar committed:
Fixed naming conventions and added some comments

SuzakuSar committed:
Fixed naming conventions to typed out numbers

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorDown.java:
```diff
@@ -0,0 +1,55 @@
+public class ElevatorDown extends Command 
+{
+  private final Elevator m_Elevator;
+  /** Creates a new ElevatorUp. */
```
fix doc

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorDown.java:
```diff
@@ -0,0 +1,55 @@
+    //intentionally empty
+  }
+
+  /**While trigger is held down applies constant voltage to move the bot down
```
docs should be relevant to this method only, not high level implementation elsewhere

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorDown.java:
```diff
@@ -0,0 +1,55 @@
+    m_Elevator.elevatorRequest(Global.MODE.VOLTAGE, ElevatorConstants.ELEVATOR_DOWN_VOLTAGE);
+  }
+
+  /**Once command ends sets motors to brake mode, making it resistant to external force
```
these explanations can be in comments but consider placing them inline instead of in the javadoc

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorDown.java:
```diff
@@ -0,0 +1,55 @@
+  @Override
+  public void end(boolean interrupted) 
+  {
+    m_Elevator.elevatorRequest(Global.MODE.BRAKE, 0);
```
we'll have to test how much of a difference this makes, but imagine transitioning from down to up, would applying a static brake here be good for the mechanism?

SuzakuSar replied:
did some further research and it might not be power efficient and could cause some wear over time so I replaced all of them with shutting off the voltage

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorLevelOne.java:
```diff
@@ -0,0 +1,57 @@
+public class ElevatorLevelOne extends Command 
+{
+   private final Elevator m_Elevator;
+  /** Creates a new ElevatorPosition1. */
```
Fix doc

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorLevelOne.java:
```diff
@@ -0,0 +1,57 @@
+    addRequirements(this.m_Elevator);
+  }
+
+  /**Will set elevator to the position called from ElevatorConstants
```
make comments inline instead of in javadoc

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorLevelOne.java:
```diff
@@ -0,0 +1,57 @@
+
+  /**Will set elevator to the position called from ElevatorConstants
+   * @param Global.MODE.POSITION sets motor to the position mode
+   * @param ElevatorConstants.ELEVATOR_POSITION_0 coral L2 position
```
there are no parameters for this method, please fix javadoc

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorLevelOne.java:
```diff
@@ -0,0 +1,57 @@
+    //intentionally empty
+  }
+
+  /**Once command ends sets motors to brake mode, making it resistant to external force
```
I'm not going to comment on this anymore, but please apply the change anywhere you see it.

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorLevelThree.java:
```diff
@@ -0,0 +1,57 @@
+public class ElevatorLevelThree extends Command 
+{
+   private final Elevator m_Elevator;
+  /** Creates a new ElevatorPosition1. */
```
??? please check your docs before you put to merge

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorLevelTwo.java:
```diff
@@ -0,0 +1,57 @@
+public class ElevatorLevelTwo extends Command 
+{
+   private final Elevator m_Elevator;
+  /** Creates a new ElevatorPosition1. */
```
???

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorUp.java:
```diff
@@ -0,0 +1,53 @@
+  @Override
+  public void execute() 
+  {
+    m_Elevator.elevatorRequest(Global.MODE.VOLTAGE, ElevatorConstants.ELEVATOR_UP_VOLTAGE);
```
why is this in execute

SuzakuSar replied:
I added it in along with another for downwards just in case it might be needed for testing with the elevator

jasonsoohoo replied:
Why is it in execute? The value is not changing so it only needs to be assigned once.

SuzakuSar replied:
ohh I get it now

SuzakuSar replied:
done

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ZeroElevator.java:
```diff
@@ -0,0 +1,58 @@
+  @Override
+  public void initialize() 
+  {
+    m_Elevator.elevatorRequest(Global.MODE.POSITION, ElevatorConstants.ELEVATOR_LEVEL_ZERO);
```
Typically when you zero the elevator you are setting the current position to be the zero position. This command should be named GoHome or something along those lines instead.

SuzakuSar committed:
Fixed naming conventions:
    Command names match up with coral levels
    Fixed Javadocs and comments with wrong info

SuzakuSar committed:
fixed method calls in up and down commands

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorDown.java:
```diff
@@ -0,0 +1,55 @@
+  public ElevatorDown(Elevator p_Elevator)
+  {
+    this.m_Elevator = p_Elevator;
+    // Use addRequirements() here to declare subsystem dependencies.
```
remove this comment

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorGoHome.java:
```diff
@@ -0,0 +1,56 @@
+  public ElevatorGoHome(Elevator p_Elevator)
+  {
+    this.m_Elevator = p_Elevator;
+    // Use addRequirements() here to declare subsystem dependencies.
```
remove

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorLevelFour.java:
```diff
@@ -0,0 +1,55 @@
+  public ElevatorLevelFour(Elevator p_Elevator) 
+  {
+    this.m_Elevator = p_Elevator;
+    // Use addRequirements() here to declare subsystem dependencies.
```
remove

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorLevelThree.java:
```diff
@@ -0,0 +1,55 @@
+  public ElevatorLevelThree(Elevator p_Elevator) 
+  {
+    this.m_Elevator = p_Elevator;
+    // Use addRequirements() here to declare subsystem dependencies.
```
remove

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorLevelTwo.java:
```diff
@@ -0,0 +1,55 @@
+  public ElevatorLevelTwo(Elevator p_Elevator) 
+  {
+    this.m_Elevator = p_Elevator;
+    // Use addRequirements() here to declare subsystem dependencies.
```
remove

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorUp.java:
```diff
@@ -0,0 +1,55 @@
+  public ElevatorUp(Elevator p_Elevator)
+  {
+    this.m_Elevator = p_Elevator;
+    // Use addRequirements() here to declare subsystem dependencies.
```
remove

jasonsoohoo commented on Big-Bad-Wolf-2025/src/main/java/frc/robot/commands/ElevatorDown.java:
```diff
@@ -0,0 +1,55 @@
+    m_Elevator.elevatorRequest(Global.MODE.VOLTAGE, ElevatorConstants.ELEVATOR_DOWN_VOLTAGE);
+  }
+
+  /**Applies voltage for Elevator to go down
```
These comments are still JavaDocs, please move them within the method instead.

jasonsoohoo replied:
And make them normal // comments

jasonsoohoo replied:
Do it for all the comments in the PR.

SuzakuSar committed:
Fixed all java docs and comments
